GLAA.Web/Controllers/PublicRegisterController.cs

```diff
@@ -0,0 +1,161 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using GLAA.Services.PublicRegister;
+using GLAA.ViewModels.PublicRegister;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Rendering;
+
+namespace GLAA.Web.Controllers
+{
+    public class PublicRegisterController : Controller
+    {
+        private readonly IPublicRegisterViewModelBuilder publicRegisterViewModelBuilder;
+        private static PublicRegisterSearchViewModel currentSearchViewModel;
+
+        public PublicRegisterController(IPublicRegisterViewModelBuilder publicRegisterViewModelBuilder)
+        {
+            this.publicRegisterViewModelBuilder = publicRegisterViewModelBuilder;
+        }
+
+        [HttpPost]
+        public IActionResult Index(PublicRegisterLicenceListViewModel publicRegisterLicenceListViewModel, string submit)
+        {
+            PublicRegisterSearchViewModel newSearchViewModel = null;
+            var searchViewModel = publicRegisterLicenceListViewModel.PublicRegisterSearchViewModel;
+
+            if (searchViewModel.AvailableCountries == null)
+                searchViewModel.AvailableCountries =
+                    publicRegisterViewModelBuilder.BuildAvailableCountries();
+
+            switch (submit)
+            {
+                case "add":
+                    newSearchViewModel = AddCountry(publicRegisterLicenceListViewModel).PublicRegisterSearchViewModel;
+                    break;
+                case "search":
+                    // we want to update teh 
+                    currentSearchViewModel = searchViewModel;
+                    searchViewModel.SearchActive = true;
+                    break;
+                default:
+                    newSearchViewModel = RemoveCountry(publicRegisterLicenceListViewModel, submit).PublicRegisterSearchViewModel;
+                    break;
+            }
+
+            if (searchViewModel.SearchActive &&
+                (searchViewModel.CountriesSelected != null && searchViewModel.CountriesSelected.Count > 0 ||
+                 !string.IsNullOrWhiteSpace(searchViewModel.BusinessName)))
+            {
+                publicRegisterLicenceListViewModel =
+                    publicRegisterViewModelBuilder.BuildSearchForLicences(currentSearchViewModel);
```

Comment:
I'd like to discuss this one. I think what I'd imagined was this it would go:

1. Post with PublicRegisterSearchViewModel to an action which would validate the model
2. Redirect to get action, which imports the viewmodel and builds a response based on that to avoid duplicate post warning if the user hits back (we have an Import and Export attribute kicking about to transfer view model via temp data so you can pull it back in the get action)

Reply:
Changed to store viewmodel in the session.  This then redirects to the search results page.  There is no validation as such at this stage.  Although perhaps there will be eventually as we need to ensure there is a search criteria defined.